pkg/apis/application/v1alpha1/types.go

```diff
@@ -1275,6 +1275,10 @@ type Cluster struct {
 	ClusterResources bool `json:"clusterResources,omitempty" protobuf:"bytes,10,opt,name=clusterResources"`
 	// Reference between project and cluster that allow you automatically to be added as item inside Destinations project entity
 	Project string `json:"project,omitempty" protobuf:"bytes,11,opt,name=project"`
+	// Labels for cluster secret metadata
+	Labels map[string]string `json:"labels,omitempty" protobuf:"bytes,12,opt,name=labels"`
+	// Annotations for cluster secret metadata
+	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,13,opt,name=annotations"`
```

Comment:
Please update Equals method in types.go as well:

https://github.com/argoproj/argo-cd/blob/bd7f8c8eda74c107b50df3908b3c6f8aff4be60d/pkg/apis/application/v1alpha1/types.go#L1281

It is used by Create API method to detect difference if cluster is already registered.

Reply:
Thanks! Updated Equals to return false on the mismatch of labels/annotations